platform/mbed_wait_api_rtos.cpp

```diff
@@ -23,26 +23,45 @@
 #include "rtos/rtos.h"
 #include "platform/mbed_critical.h"
 #include "platform/mbed_power_mgmt.h"
+#include "platform/mbed_error.h"
 
 void wait(float s)
 {
-    wait_us(s * 1000000.0f);
+    if (s >= 0.01f) {
+        wait_ms(s * 1000.0f);
+    } else {
+        wait_us(s * 1000000.0f);
+    }
 }
 
+/*  The millisec value specifies the number of timer ticks and is therefore an upper bound.
+ *  The exact time delay depends on the actual time elapsed since the last timer tick.
+ *  The actual time delay may be up to one timer tick less.
+ */
 void wait_ms(int ms)
 {
-    wait_us(ms * 1000);
+    if (core_util_are_interrupts_enabled()) {
+        ThisThread::sleep_for(ms);
+    } else {
+        MBED_WARNING(MBED_MAKE_ERROR(MBED_MODULE_PLATFORM, MBED_ERROR_INVALID_OPERATION),
+                    "Deprecated behavior: milli-sec delay should not be used in interrupt.\n");
+        wait_us(ms * 1000);
+    }
 }
 
 void wait_us(int us)
 {
+    if (us > 10000) {
```

Comment:
@kjbracey-arm - 0.1s / 0.01s. I would like to keep it same for https://github.com/ARMmbed/mbed-os/pull/8189/files#diff-5a106edf0944ae7ba2d1bf95153fb0a0R30

Reply:
Yes, 0.01s makes more sense. I think I probably meant to type that.

Reply:
0.01s seems fine